File: Data Structures and Algorithms/Sword refers to offer/C++/14.cpp
//1
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
        if(pListHead==NULL)
            return NULL;
        ListNode* left,*right;
		left=pListHead;
		right=pListHead;
		while(k&&right!=NULL)
		{
			right=right->next;
            k--;
		} 
        if(right==NULL&&k>0)
            return NULL;
		while(right!=NULL)
		{
			right=right->next;
			left=left->next; 
		}
		return left;
    }
};

File: Data Structures and Algorithms/Sword refers to offer/Python/14.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def FindKthToTail(self, head, k):
        # write code here
        num=k
        left=head
        right=head
        while num>0 and right!=None:
            right=right.next
            num-=1
        if num>0:
            return None
        while right!=None:
            left=left.next
            right=right.next
        return left
